services/web/server/src/simcore_service_webserver/dynamic_scheduler/api.py

```diff
@@ -81,14 +105,32 @@ async def stop_dynamic_service(
         if progress:
             await stack.enter_async_context(progress)
 
-        settings: DynamicSchedulerSettings = get_plugin_settings(app)
+        rpc_client = get_rabbitmq_rpc_client(app)
         await services.stop_dynamic_service(
-            get_rabbitmq_rpc_client(app),
-            dynamic_service_stop=dynamic_service_stop,
-            timeout_s=int(
+            rpc_client, dynamic_service_stop=dynamic_service_stop
+        )
+
+        # wait for the service to be stopped, until it becomes idle
+        settings: DynamicSchedulerSettings = get_plugin_settings(app)
+        async for attempt in AsyncRetrying(
+            wait=wait_fixed(1.0),
+            stop=stop_after_delay(
                 settings.DYNAMIC_SCHEDULER_STOP_SERVICE_TIMEOUT.total_seconds()
             ),
-        )
+            before_sleep=partial(
```

Comment:
hmm you are maybe looking for before_sleep_log?

Reply:
no, this is correct i want a custom function here